utils/cache_manager.py

```
@@ -0,0 +1,401 @@
+"""
+Unified Cache Manager for centralized cache management and monitoring.
+
+This module provides a centralized interface for managing all caching systems
+in the Zen MCP Server, including token estimation, schema generation, and
+model validation caches.
+"""
+
+import logging
+import threading
+import time
+from typing import Any, Dict, List, Optional
```

Ruff (0.11.9)

`typing.Dict` is deprecated, use `dict` instead

(UP035)